### ISSUE TITLE
Policy Violations do not show up in cloned projects

### ISSUE THREAD

**Issue by ofg** [DependencyTrack/dependency-track#2875]
### Current Behavior

Cloning a project which has license policy violations does not clone the license policy violations. In fact, policy violations is completely empty.

May be related to #2640

### Steps to Reproduce

1. create a project with 
```
project_name="test_project"
project_version="0.0.1"
filename="test-2.sbom"
curl -L -sS  \
    -X POST "${DEPENDENCY_TRACK_URL}/api/v1/bom" \
    -H "X-API-Key: $DEPENDENCY_TRACK_API_TOKEN" \
    -H "Content-Type: multipart/form-data" \
    -H "Accept: application/json" \
    -F "autoCreate=true" \
    -F "projectName=$project_name" \
    -F "projectVersion=$project_version" \
    -F "bom=@$filename"
```
[test-2.sbom.zip](https://github.com/DependencyTrack/dependency-track/files/11968860/test-2.sbom.zip)

It shows the following license policy violation:
![image](https://github.com/DependencyTrack/dependency-track/assets/7761543/46c05083-13c6-47aa-ac2d-acfccaa670ee)

An it returns the following metrics calling `/api/v1/metrics/project/<prj uuid>/current`:
```
{
...
  "components": 5,
...
  "policyViolationsFail": 0,
  "policyViolationsWarn": 1,
  "policyViolationsInfo": 0,
  "policyViolationsTotal": 1,
  "policyViolationsAudited": 0,
  "policyViolationsUnaudited": 1,
  "policyViolationsSecurityTotal": 0,
  "policyViolationsSecurityAudited": 0,
  "policyViolationsSecurityUnaudited": 0,
  "policyViolationsLicenseTotal": 1,
  "policyViolationsLicenseAudited": 0,
  "policyViolationsLicenseUnaudited": 1,
...
}
```

2. Clone the project with
```
# src_prj_uuid is retrieved from project created by step 1
new_project_version="0.0.2"
curl -L -sS -w "%{http_code}" -o "$DT_RESPONSE_FILE" \
    -X PUT "${DEPENDENCY_TRACK_URL}/api/v1/project/clone" \
    -d "{ \
      \"project\": \"$src_prj_uuid\", \
      \"version\": \"$new_project_version\", \
      \"includeTags\": true, \
      \"includeProperties\": false, \
      \"includeComponents\": true, \
      \"includeServices\": true, \
      \"includeAuditHistory\": true, \
      \"includeACL\": true \
    }" \
    -H "X-API-Key: $DEPENDENCY_TRACK_API_TOKEN" \
    -H "Accept: application/json" \
    -H "Content-Type: application/json"
```
The new project version 0.0.2 does not show any license policy violation:
![image](https://github.com/DependencyTrack/dependency-track/assets/7761543/ac96f907-976b-471c-83e8-9bb6c78769f4)

Furthermore, it also does not show the "Last BOM Import" nor the "Last Measurement" timestamp:
![image](https://github.com/DependencyTrack/dependency-track/assets/7761543/327b2d9e-98aa-43fc-bb59-aba029d50ac9)

And the returned metrics calling `/api/v1/metrics/project/<prj uuid>/current` does return http status code 200 but no response body.

3. Refresh project metrics with `/api​/v1​/metrics​/project​/<prj uuid>​/refresh` does set a new "Last Measurement" timestamp, and calling `metrics/.../current` now returns a response body - but it does not contain the correct policy violation numbers - and also the UI still shows no license policy violations:
```
...
  "components": 5,
...
  "policyViolationsFail": 0,
  "policyViolationsWarn": 0,
  "policyViolationsInfo": 0,
  "policyViolationsTotal": 0,
  "policyViolationsAudited": 0,
  "policyViolationsUnaudited": 0,
  "policyViolationsSecurityTotal": 0,
  "policyViolationsSecurityAudited": 0,
  "policyViolationsSecurityUnaudited": 0,
  "policyViolationsLicenseTotal": 0,
  "policyViolationsLicenseAudited": 0,
  "policyViolationsLicenseUnaudited": 0,
...
}
```

4. Repeat step 1 to 3 but accept the license policy violation in step 1. This is also not cloned.

### Expected Behavior

After cloning a project, I would expect to see the same policy violations as within the original project.

I would also expect to see the audit trail in case a license policy violation had been ACCEPTED / REJECTED in the original project.

### Dependency-Track Version

4.8.2

### Dependency-Track Distribution

Container Image

### Database Server

PostgreSQL

### Database Server Version

14.7 (AWS RDS PostgreSQL instance)

### Browser

N/A

### Checklist

- [X] I have read and understand the [contributing guidelines](https://github.com/DependencyTrack/dependency-track/blob/master/CONTRIBUTING.md#filing-issues)
- [X] I have checked the [existing issues](https://github.com/DependencyTrack/dependency-track/issues) for whether this defect was already reported

**Comment by nscuro** [comment]
Note to self: Policy violations are currently not considered in the project cloning logic:

https://github.com/DependencyTrack/dependency-track/blob/d4f1e8ef631ed4c22ff1a483f277713e22c09697/src/main/java/org/dependencytrack/persistence/ProjectQueryManager.java#L645C17-L652

Same goes for the violation analysis trail:

https://github.com/DependencyTrack/dependency-track/blob/d4f1e8ef631ed4c22ff1a483f277713e22c09697/src/main/java/org/dependencytrack/persistence/ProjectQueryManager.java#L657-L687

**Comment by rkg-mm** [comment]
I think this issue here, #2640 and #1732  are duplicates @nscuro  am I right?
we are currently implementing a solution for this, can we ensure these are the same requests so we can finish them all?

**Comment by nscuro** [comment]
@rkg-mm Looks like that to me too, yes. I'll close the others as duplicates so we can track it here.

**Comment by nscuro** [comment]
This was implemented with #3248.
